### ISSUE TITLE
Feature request: expose isCurrent & isPartiallyCurrent as public API helpers

### ISSUE THREAD

**Issue by ssured** [reach/router#107]
First of all, thanks for the lib! 

## Context

I'm making a collapsible navigation menu. The menu has groups with inner links.
Each group is collapsible, but should be forced open if any of the inner links is active.

With the current API I do not see a proper solution. To solve this I see two solutions.

### 1) expand the Match component to accept multiple paths
```javascript
  <Match path={['/one','/two/three']}>({match}) => {....})</Match>
```

### 2) expose isCurrent and isPartiallyCurrent as public API helpers

https://github.com/reach/router/blob/054187e17795c5c2df289109456ce42c48f49343/src/index.js#L377-L379 should be public

Would it be possible to add this to the library?

**Comment by ricardobrandao** [comment]
`Link` components already have a `getProps` property that receives those properties. Please refer to the [example in the documentation](https://reach.tech/router/example/active-links).

**Comment by ssured** [comment]
@ricardobrandao, true, that's giving the info on the child level. What I need is to have a check for the parent level, to see if any of it's children is active. I do not want to use the routing hierarchy, but just match against a set of paths if any of these paths `isCurrent`

**Comment by jedmao** [comment]
This would also be useful for [Linaria](https://linaria.now.sh/) or [styled components](https://www.styled-components.com/):

```ts
import { styled } from 'linaria/react'
import { Link, LinkProps } from '@reach/router'
import React, { FC } from 'react'

const RedLinkIfCurrent: React.FC<LinkProps<any>>(Link)`
  color: ${props => props.isCurrent ? 'red' : 'black' };
`
//                            ^-------- doesn't work
```
